File: 0645-set-mismatch/0645-set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int n = nums.size();
        int missing = n*(n+1)/2, duplicate = -1;

        // extra array

        vector <int> pres(n,0);

        for(int x:nums)
            if(pres[x-1]==0){pres[x-1] = x;missing-=x;}
            else duplicate = x;

        return {duplicate, missing};
    }
};

File: 0645-set-mismatch/0645-set-mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        n = len(nums)
        s = n*(n+1)//2
        t = 0
        pre = [0]*n

        for x in nums:
            if pre[x-1]==0:
                pre[x-1] = 1
                s-=x
            else:
                t = x
            # print(x,pre,s,t)
        
        return [t,s]
